diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05001ad..8f07b8c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,10 @@ include (DetectCPPZMQVersion)
 project(cppzmq VERSION ${DETECTED_CPPZMQ_VERSION})

 if (NOT TARGET libzmq AND NOT TARGET libzmq-static)
+  add_library(libzmq UNKNOWN IMPORTED)
+  set_target_properties(libzmq PROPERTIES
+            IMPORTED_LOCATION ${ZeroMQ_LIBRARIES}
+            INTERFACE_INCLUDE_DIRECTORIES ${ZeroMQ_INCLUDE_DIRS})
   find_package(ZeroMQ QUIET)

   # libzmq autotools install: fallback to pkg-config
